[UX]: live table when provisioning dstack configuration runs #1978

Tob-iee opened the pull request:
fixes #1951 @jvstme, yeah, you are right I didn't reflect the changes after merging dstackai:master into my branch. I re-opened it. Can you have another look? it should work fine

Tob-iee committed:
[UX]: live table when provisioning dstack configuration runs

Tob-iee committed:
Merge branch 'dstackai:master' into issue_1951_live-table-for-runs

jvstme commented on src/dstack/_internal/cli/services/configurators/run.py:
```diff
@@ -221,6 +225,9 @@ def apply_configuration(
                     exit(1)
+                    break
+                time.sleep(LIVE_TABLE_PROVISION_INTERVAL_SECS)
+                run.refresh()
```
These lines seem to be redundant:
- `break` will never be reached after `exit(1)` on line 227
- `time.sleep` is redundant because the decision to reattach is already made on line 216
- `run.refresh` is redundant because the run was already refreshed on line 213

jvstme committed:
Minor enhancements

- Remove redundant instructions when reattaching
- Add empty lines before and after the table
- Keep updating the table while the run is
  `terminating`. This provides faster feedback and
  more details for troubleshooting
- When provisioning finishes, update the table one
  more time so that it shows the relevant run
  status, e.g. `running`

jvstme commented:
@Tob-iee, thanks for reopening the PR. This version looks and works great. I've made some minor tweaks and will merge the PR soon. Thanks for contributing!
